shared_model/cryptography/ed25519_ursa_impl/crypto_provider.cpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "cryptography/ed25519_ursa_impl/crypto_provider.hpp"
+#include "ursa_crypto.h"
+#include <iostream>
+
+namespace shared_model {
+  namespace crypto {
+    Signed CryptoProviderEd25519Ursa::sign(const Blob &blob,
+                                           const Keypair &keypair) {
+      ByteBuffer signature;
+      ExternError err;
+
+      if (!ursa_ed25519_sign(
+        blob.blob().data(),
+        blob.blob().size(),
+        keypair.privateKey().blob().data(),
+        keypair.privateKey().blob().size(),
+        &signature,
+        &err
+      )) {
+        // handle error
+        free(err.message);
```

Comment:
Please check that `err.message` was allocated with `malloc` functions family.
If that memory was allocated with `new`, then it should be deallocated with `delete`, otherwise that may lead to undefined behavior.